azure-mgmt-scheduler/src/main/java/com/microsoft/azure/management/scheduler/Job.java

```diff
@@ -93,33 +98,217 @@
+             * <p>
+             * The recurrence object supports the elements frequency, interval, endTime, count, and schedule. If recurrence
+             *   is defined, frequency is required; the other elements of recurrence are optional.
              *
-             * @param jobRecurrence the start time
+             * @param jobRecurrence the reccurence object
              * @return the next stage of the definition
              */
             WithAction withRecurrence(JobRecurrence jobRecurrence);
 
             /**
-             * Specifies the action details of the job.
+             * Specifies the interval for the frequency that determines how often the job will run.
+             * <p>
+             * For example, if interval is 3 and frequency is "week", the job recurs every 3 weeks. Azure Scheduler
+             *   supports a maximum interval of 18 months for monthly frequency, 78 weeks for weekly frequency,
+             *   or 548 days for daily frequency. For hour and minute frequency, the supported range is 1 <= interval <= 1000.
              *
-             * @param jobAction the action details
+             * @param interval a positive integer and denotes the interval for the frequency that determines how often the job will run
              * @return the next stage of the definition
              */
-            WithCreate withAction(JobAction jobAction);
+            WithRecurrenceFrequencyAndSchedule runningEvery(int interval);
+
+            /**
+             * Specifies the recurrence of the job to be every minute.
+             *
+             * @return the next stage of the definition
+             */
+            WithEndTime runningEveryMinute();
+
+            /**
+             * Specifies the recurrence of the job to be every hour.
+             *
+             * @return the next stage of the definition
+             */
+            WithEndTime runningHourly();
+
+            /**
+             * Specifies the recurrence of the job to be every day.
+             *
+             * @return the next stage of the definition
+             */
+            WithAdvanceMinutesAndHoursSchedule runningDaily();
+
+            /**
+             * Specifies the recurrence  of the job to be every week.
+             *
+             * @return the next stage of the definition
+             */
+            WithAdvanceDaysOfWeekSchedule runningWeekly();
+
+            /**
+             * Specifies the recurrence of the job to be every month.
+             *
+             * @return the next stage of the definition
+             */
+            WithAdvanceDaysOfMonthSchedule runningMonthly();
+        }
+
+        /**
+         * The stage of the job definition for an Azure Scheduler allowing to specify the frequency unit.
+         */
+        interface WithRecurrenceFrequencyAndSchedule {
+            /**
+             * Specifies the recurrence of the job to be every minute.
```

Comment:
I don;t think this text is correct. Shouldn't it be more like: 
"Selects minutes as the time unit for the number of occurrences."